docs/design-proposals/cloud-edge-communication/cloud-edge-communication-framework.md

```diff
@@ -30,28 +30,67 @@ edge server -----> Proxy A(in the edge) <----------> Proxy B(in the cloud) -----
 Usually, there are many kinds of services communication between the cloud and edge. For example, the communication may be rest API calls, gprc, message call(kafka/rabbitmq) or something else.
 
-The proxy may work like a nginx(work in network layer 4 or 7), to support all kinds of the communications in the edge cloud. And the proxy A and Proxy B need to communicat to each other with mTLS for security.
+How to implement the proxy?
 
-Problem of proposal A(to be supplemented)
+Option 1: The proxy adapt the communication protocols(network layer 7).
+
+For example, if the proxy want to support rest calls(http/https) between the cloud and the edge. The proxy works as a http/https server, it can proxy the rest calls to the destination site.
+And the proxy can distinguish the target service from the request url parsing to support multiple backend http servers.
+For https request, the proxy play the middle man role to do the traffic encryption, decryption and forward.
+
+If the proxy want to support message(rabbitmq/kafka) between the cloud and the edge. The proxy works as a amqp/kafka server, it can proxy the message call/connection to the destination site, it need to understand the amqp/kafka protocol, to do the connection and request forward.
+And also there are encryption/decryption problems(MitM). It is hard to support all the communcation protocols in this way.
+
+So if the rest call(http/https request) is the only communication type between the cloud and the edge, maybe it is a simple/good choice. 
+
+Option 2: The proxy works at network layer 4, do the udp/tcp proxy.
+
+For example, we want to support rest calls(http/https), message queue connection(rabbitmq/kafka), and something else based on the udp protocol.
+The proxy works as a udp server, and it forward the udp connection/traffic to the destination site.
+
+It works just like a nginx, but there are some differences:
+1. It is not proxy the traffic to the destination service directly, but to the other proxy in other site
+2. It does not work only for one or two services, and with static configuration. It needs to support multiple services at the same time, and the services can be dynamic add or remove.
+It needs to solve the port allocation and traffic encryption/decryption problems
+
+![](Proposal-B4.png) 
 
 ## Proposal B
 
-We want to implement it based on network layer 3, and we also need to solve the ip address conflict and security problems.
+We want to implement the communication framework based on the network layer 3, and also we need to solve the ip address conflict and security problems. We can use ip translation to do this. 
+
+We need to add the gateway/proxy for the edge site and the cloud for the communication, we call it edgegateway in each edge site, and in addition to communication, cloudgateway in responsible for central control. We don't expose the network to the cloud or the edge site(cause too much attach surface), we expose the services to the gateway.  
```

Comment:
in responsible --> is responsible

attach surface --> attack surface

so in short. Both cloud and edge sides have gateways. Services in both sides are behind the gateways.